app/models/store.rb

```diff
@@ -22,10 +22,11 @@ def active_items
     items.active
   end
 
-  def update_status(status)
-    store_admin_role = Role.find_by(name: "Store Admin")
+  def update_status(status, initial_req = nil)
+    if initial_req
```

Comment:
Unless the store's status is being changed for an initial request, "initial_req" will have the value of nil.

Reply:
I really like this refactor so that we're only updating the store's user_roles if it's new.